### PR TITLE
[Chore] increase number of workers for the tests.

### PR DESCRIPTION
**Pull request by sayakpaul**
Additionally, this PR changes the `docker-cpu` runners to a a better one.

### CODE REVIEW

**Comment by sayakpaul** [comment]
@DN6 any idea why more runners would incur time-outs?

File: .github/workflows/pr_tests.yml [hunk 1]
```diff
@@ -130,7 +130,7 @@ jobs:
       if: ${{ matrix.config.framework == 'pytorch_pipelines' }}
       run: |
         python -m venv /opt/venv && export PATH="/opt/venv/bin:$PATH"
-        python -m pytest -n 2 --max-worker-restart=0 --dist=loadfile \
+        python -m pytest -n 4 --max-worker-restart=0 --dist=loadfile \
```
Comment (DN6) [inline .github/workflows/pr_tests.yml:133]: If the runner has 32 cores available for pipelines why not bump this up further? To 8?

Comment (sayakpaul) [reply .github/workflows/pr_tests.yml]: Done in 9afd3cca. 
🤞
Let's see

**Comment by sayakpaul** [comment]
Nice, we're down to 7 minutes now.

**Review by DN6 — APPROVED** [review]

**Pull request merged**